akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -26,9 +26,10 @@ private[akka] trait DeathWatch { this: ActorCell ⇒
       if (a != self && !watchingContains(a)) {
         maintainAddressTerminatedSubscription(a) {
           a.sendSystemMessage(Watch(a, self)) // ➡➡➡ NEVER SEND THE SAME SYSTEM MESSAGE OBJECT TO TWO ACTORS ⬅⬅⬅
-          watching = watching.updated(a, None)
+          updateWatching(a, None)
         }
-      }
+      } else if (watchingContains(a))
```

Comment:
Not a big thing, but could we do this check only once with nested ifs perhaps?

Reply:
Ok (probably not critical as it's only on a fallback branch anyway, would only match when the watchee is `self`)

Reply:
Yeah, it was more of a eye-sore issue than performance related :)